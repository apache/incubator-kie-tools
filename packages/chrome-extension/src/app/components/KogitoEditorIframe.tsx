/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Router } from "@kogito-tooling/core-api";
import * as React from "react";
import { useContext, useEffect, useRef } from "react";
import { GlobalContext } from "./GlobalContext";
import { EnvelopeBusOuterMessageHandler } from "@kogito-tooling/microeditor-envelope-protocol";
import { GitHubDomElements } from "../../GitHubDomElementsFactory";
import { runScriptOnPage } from "../utils";

const githubCodeMirrorEditorSelector = `.file-editor-textarea + .CodeMirror`;

let prevGlobalState: any;
let polling: any;

export function KogitoEditorIframe(props: {
  openFileExtension: string;
  githubDomElements: GitHubDomElements;
  router: Router;
}) {
  const [globalState, setGlobalState] = useContext(GlobalContext);
  const iframeRef = useRef<HTMLIFrameElement | null>(null);

  const envelopeBusOuterMessageHandler = new EnvelopeBusOuterMessageHandler(
    {
      postMessage: msg => {
        if (iframeRef.current && iframeRef.current.contentWindow) {
          iframeRef.current.contentWindow.postMessage(msg, props.router.getTargetOrigin());
        }
      }
    },
    self => ({
      pollInit() {
        self.request_initResponse(window.location.origin);
      },
      receive_languageRequest() {
        self.respond_languageRequest(props.router.getLanguageData(props.openFileExtension));
      },
      receive_contentResponse(content: string) {
        runScriptOnPage(
          `document.querySelector("${githubCodeMirrorEditorSelector}").CodeMirror.setValue('${content}')`
        );
      },
      receive_contentRequest() {
        self.respond_contentRequest(props.githubDomElements.githubContentTextArea().value);
        startPollingForChangesOnDiagram();
      },
      receive_setContentError() {
        //TODO: Display a nice message with explanation why "setContent" failed
        console.info("Set content error");
      },
      receive_dirtyIndicatorChange(isDirty: boolean) {
        //TODO: Perhaps show window.alert to warn that the changes were not saved?
        console.info(`Dirty indicator changed to ${isDirty}`);
      },
      receive_ready() {
        console.info(`Editor is ready`);
        setGlobalState({ ...globalState, textModeEnabled: true });
      }
    })
  );

  function startPollingForChangesOnDiagram() {
    if (polling) {
      return;
    }

    polling = setInterval(() => envelopeBusOuterMessageHandler.request_contentResponse(), 1500);
  }

  function stopPollingForChangesOnDiagram() {
    clearInterval(polling);
    polling = undefined;
  }

  useEffect(
    () => {
      if (globalState.textMode) {
        stopPollingForChangesOnDiagram();
        envelopeBusOuterMessageHandler.request_contentResponse();
      } else if (prevGlobalState && prevGlobalState.textMode !== globalState.textMode) {
        envelopeBusOuterMessageHandler.respond_contentRequest(props.githubDomElements.githubContentTextArea().value);
        startPollingForChangesOnDiagram();
      }

      prevGlobalState = globalState;
    },
    [globalState]
  );

  useEffect(() => {
    const listener = (msg: MessageEvent) => envelopeBusOuterMessageHandler.receive(msg.data);
    window.addEventListener("message", listener, false);
    envelopeBusOuterMessageHandler.startInitPolling();

    return () => {
      envelopeBusOuterMessageHandler.stopInitPolling();
      window.removeEventListener("message", listener);
      stopPollingForChangesOnDiagram();
    };
  }, []);

  return (
    <>
      <iframe
        ref={iframeRef}
        id={"kogito-iframe"}
        className={globalState.fullscreen ? "fullscreen" : "not-fullscreen"}
        src={props.router.getRelativePathTo("envelope/index.html")}
      />
    </>
  );
}
