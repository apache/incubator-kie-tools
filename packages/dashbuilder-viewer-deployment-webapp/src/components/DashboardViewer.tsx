/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { EmbeddedEditorFile } from "@kie-tools-core/editor/dist/channel";
import { EmbeddedEditor, useEditorRef } from "@kie-tools-core/editor/dist/embedded";
import { useState, useMemo, useCallback } from "react";
import {
  ChannelType,
  EditorEnvelopeLocator,
  EnvelopeMapping,
  EnvelopeContentType,
} from "@kie-tools-core/editor/dist/api";
import { useCancelableEffect } from "@kie-tools-core/react-hooks/dist/useCancelableEffect";
import { Dashboard } from "../data";
import { extractExtension } from "@kie-tools-core/workspaces-git-fs/dist/relativePath/WorkspaceFileRelativePathParser";
import { basename } from "path";

interface DashboardViewerProps {
  dashboard: Dashboard;
}

export function DashboardViewer(props: DashboardViewerProps) {
  const { editor, editorRef } = useEditorRef();
  const [embeddedEditorFile, setEmbeddedEditorFile] = useState<EmbeddedEditorFile>();

  const editorEnvelopeLocator = useMemo(
    () =>
      new EditorEnvelopeLocator(window.location.origin, [
        new EnvelopeMapping({
          type: "dash",
          filePathGlob: "**/*.dash.+(yml|yaml)",
          resourcesPathPrefix: "",
          envelopeContent: { type: EnvelopeContentType.PATH, path: "dashbuilder-viewer-envelope.html" },
        }),
      ]),
    []
  );

  useCancelableEffect(
    useCallback(
      ({ canceled }) => {
        fetch(props.dashboard.uri)
          .then((response) => response.text())
          .then((content) => {
            if (canceled.get()) {
              return;
            }

            const extension = extractExtension(props.dashboard.uri);
            const filename = basename(props.dashboard.uri, `.${extension}`);

            setEmbeddedEditorFile({
              path: props.dashboard.uri,
              getFileContents: async () => content,
              isReadOnly: true,
              fileExtension: extension,
              fileName: filename,
            });
          });
      },
      [props.dashboard.uri]
    )
  );

  return (
    <div style={{ height: "100%" }}>
      {embeddedEditorFile && (
        <EmbeddedEditor
          channelType={ChannelType.EMBEDDED}
          editorEnvelopeLocator={editorEnvelopeLocator}
          locale={"en"}
          ref={editorRef}
          file={embeddedEditorFile}
        />
      )}
    </div>
  );
}
