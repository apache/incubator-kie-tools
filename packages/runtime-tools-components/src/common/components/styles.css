/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

.kogito-components-common__load-more {
  margin-bottom: var(--pf-global--spacer--lg) !important;
  text-align: center;
}

.pf-c-spinner .pf-m-md .kogito-components-common__load-more-spinner {
  overflow: hidden;
  margin-left: 5px;
}

.kogito-components-common__form-footer-padding-top {
  padding-top: 20px;
}

.kogito-request-data-editor-container {
  border: #d2d2d2 solid 1px;
  height: 350px;
  padding: 5px;
}
