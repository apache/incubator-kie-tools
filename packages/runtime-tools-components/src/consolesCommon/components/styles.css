.kogito-consoles-common--aboutModalBox .pf-c-about-modal-box__hero {
  background-attachment: unset;
}

.kogito-consoles-common--PageLayout .pf-c-page__header-brand-link .pf-c-brand {
  --pf-c-page__header-brand-link--c-brand--MaxHeight: 42px;
}
