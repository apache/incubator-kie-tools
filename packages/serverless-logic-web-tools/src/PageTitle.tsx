/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { APP_NAME } from "./AppConstants";

/**
 *
 * Sets the title of the page by concatenating the given title sections and append the application name.
 * @param titleSections An array of strings representing the sections of the title in the order of [parent, children]
 * @returns
 */
export function setPageTitle(titleSections: string[]): void {
  document.title = [APP_NAME, ...titleSections].reverse().join(" | ");
}
