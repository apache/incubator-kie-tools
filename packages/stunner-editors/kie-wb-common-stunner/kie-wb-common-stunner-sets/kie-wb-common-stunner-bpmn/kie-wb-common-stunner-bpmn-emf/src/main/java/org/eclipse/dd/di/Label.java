/**
 * <copyright>
 * 
 * Copyright (c) 2010 SAP AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *    Reiner Hille-Doering (SAP AG) - initial API and implementation and/or initial documentation
 * 
 * </copyright>
 */
package org.eclipse.dd.di;

import org.eclipse.dd.dc.Bounds;

/**
 * <!-- begin-user-doc -->
 * A representation of the model object '<em><b>Label</b></em>'.
 * <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link org.eclipse.dd.di.Label#getBounds <em>Bounds</em>}</li>
 * </ul>
 *
 * @see org.eclipse.dd.di.DiPackage#getLabel()
 * @model abstract="true"
 *        extendedMetaData="name='Label' kind='elementOnly'"
 * @generated
 */
public interface Label extends Node {
	/**
	 * Returns the value of the '<em><b>Bounds</b></em>' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @return the value of the '<em>Bounds</em>' containment reference.
	 * @see #setBounds(Bounds)
	 * @see org.eclipse.dd.di.DiPackage#getLabel_Bounds()
	 * @model containment="true" ordered="false"
	 *        extendedMetaData="kind='element' name='Bounds' namespace='http://www.omg.org/spec/DD/20100524/DC'"
	 * @generated
	 */
	Bounds getBounds();

	/**
	 * Sets the value of the '{@link org.eclipse.dd.di.Label#getBounds <em>Bounds</em>}' containment reference.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the new value of the '<em>Bounds</em>' containment reference.
	 * @see #getBounds()
	 * @generated
	 */
	void setBounds(Bounds value);

} // Label
