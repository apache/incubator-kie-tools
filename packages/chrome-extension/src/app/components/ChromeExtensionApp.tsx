/*
 * Copyright 2019 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { GitHubDomElements } from "../../GitHubDomElementsFactory";
import * as ReactDOM from "react-dom";
import * as React from "react";
import { useEffect, useLayoutEffect, useState } from "react";
import { Router } from "@kogito-tooling/core-api";
import { GlobalContext } from "./GlobalContext";
import { KogitoEditorIframe } from "./KogitoEditorIframe";
import { FullScreenToolbar } from "./FullScreenToolbar";
import { Toolbar } from "./Toolbar";

export function ChromeExtensionApp(props: {
  githubDomElements: GitHubDomElements;
  openFileExtension: string;
  router: Router;
}) {
  const [globalState, setGlobalState] = useState({ fullscreen: false, textMode: false, textModeEnabled: false });

  useLayoutEffect(
    () => {
      if (!globalState.fullscreen) {
        props.githubDomElements.iframeFullscreenContainer().classList.add("hidden");
      } else {
        props.githubDomElements.iframeFullscreenContainer().classList.remove("hidden");
      }
    },
    [globalState]
  );

  useEffect(
    () => {
      if (globalState.textMode) {
        props.githubDomElements.githubTextEditor().classList.remove("hidden");
        props.githubDomElements.iframeContainer().classList.add("hidden");
      } else {
        props.githubDomElements.githubTextEditor().classList.add("hidden");
        props.githubDomElements.iframeContainer().classList.remove("hidden");
      }
    },
    [globalState]
  );

  return (
    <GlobalContext.Provider value={[globalState, setGlobalState]}>
      {ReactDOM.createPortal(<Toolbar />, props.githubDomElements.toolbarContainer())}

      {globalState.fullscreen &&
        ReactDOM.createPortal(<FullScreenToolbar />, props.githubDomElements.iframeFullscreenContainer())}

      {ReactDOM.createPortal(
        <KogitoEditorIframe
          openFileExtension={props.openFileExtension}
          router={props.router}
          githubDomElements={props.githubDomElements}
        />,
        globalState.fullscreen
          ? props.githubDomElements.iframeFullscreenContainer()
          : props.githubDomElements.iframeContainer()
      )}
    </GlobalContext.Provider>
  );
}
