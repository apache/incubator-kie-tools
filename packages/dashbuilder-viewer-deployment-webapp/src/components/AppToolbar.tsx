/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { I18nHtml } from "@kie-tools-core/i18n/dist/react-components";
import { extractExtension } from "@kie-tools-core/workspaces-git-fs/dist/relativePath/WorkspaceFileRelativePathParser";
import { Masthead, MastheadBrand, MastheadMain } from "@patternfly/react-core/dist/js/components/Masthead";
import { PageHeaderToolsItem } from "@patternfly/react-core/dist/js/components/Page";
import { Text, TextContent, TextVariants } from "@patternfly/react-core/dist/js/components/Text";
import { Tooltip } from "@patternfly/react-core/dist/js/components/Tooltip";
import { Flex, FlexItem } from "@patternfly/react-core/dist/js/layouts/Flex";
import { HelpIcon } from "@patternfly/react-icons/dist/js/icons/help-icon";
import { basename } from "path";
import { useCallback, useMemo, useState } from "react";
import { useHistory } from "react-router";
import { useAppI18n } from "../i18n";
import { routes } from "../routes";
import { Dashboard } from "../data";
import {
  Dropdown,
  DropdownItem,
  DropdownPosition,
  DropdownToggle,
} from "@patternfly/react-core/dist/js/components/Dropdown";
import { useApp } from "../context/AppContext";

interface Props {
  dashboard: Dashboard;
}

export function AppToolbar(props: Props) {
  const app = useApp();
  const history = useHistory();
  const { i18n } = useAppI18n();
  const [modelDropdownOpen, setModelDropdownOpen] = useState(false);

  const openDashboard = useCallback(
    (uri: string) => {
      history.push({
        pathname: routes.dashboard.path({ filePath: uri }),
      });
    },
    [history]
  );

  const modelDropdownItems = useMemo(() => {
    return app.dashboards
      .map((db) => db.uri)
      .filter((uri) => uri !== props.dashboard.uri)
      .sort((a, b) => a.localeCompare(b))
      .map((uri, idx) => (
        <DropdownItem
          key={`dashboard-toolbar-dropdown-item-${idx}`}
          component="button"
          onClick={() => openDashboard(uri)}
        >
          {basename(uri)}
        </DropdownItem>
      ));
  }, [app.dashboards, openDashboard, props.dashboard.uri]);

  const filename = useMemo(() => {
    const fullFilename = basename(props.dashboard.uri);
    const maxSize = 35;
    const extension = extractExtension(fullFilename);
    const name = fullFilename.replace(`.${extension}`, "");

    if (fullFilename.length < maxSize) {
      return fullFilename;
    }

    return `${name.substring(0, maxSize - extension.length)}...${extension}`;
  }, [props.dashboard.uri]);

  return (
    <Masthead aria-label={"Page header"} display={{ default: "stack" }} className="app--masthead">
      <MastheadMain style={{ justifyContent: "space-between" }}>
        <PageHeaderToolsItem className={"pf-l-flex"}>
          <MastheadBrand
            onClick={() => history.push({ pathname: routes.root.path({}) })}
            style={{ textDecoration: "none" }}
          >
            <TextContent>
              <Text component={TextVariants.h1}>Dashbuilder Deployment</Text>
            </TextContent>
          </MastheadBrand>
        </PageHeaderToolsItem>
        <PageHeaderToolsItem>
          <Tooltip
            className="app--masterhead__disclaimer"
            position="bottom-end"
            key="disclaimer-tooltip"
            content={<I18nHtml>{i18n.masthead.disclaimer.description}</I18nHtml>}
          >
            <TextContent>
              <Text component={TextVariants.h5}>
                {i18n.masthead.disclaimer.title}
                <HelpIcon className="app--masterhead__disclaimer-icon" />
              </Text>
            </TextContent>
          </Tooltip>
        </PageHeaderToolsItem>
        <Flex justifyContent={{ default: "justifyContentFlexEnd" }}>
          <FlexItem>
            <PageHeaderToolsItem>
              {app.dashboards.length > 1 ? (
                <Dropdown
                  onSelect={() => setModelDropdownOpen(false)}
                  toggle={
                    <DropdownToggle
                      className="app--toolbar-dropdown-hoverable-dark"
                      onToggle={(isOpen) => setModelDropdownOpen(isOpen)}
                    >
                      {filename}
                    </DropdownToggle>
                  }
                  isOpen={modelDropdownOpen}
                  position={DropdownPosition.right}
                  dropdownItems={modelDropdownItems}
                />
              ) : (
                <TextContent>
                  <Text component={TextVariants.h5}>{filename}</Text>
                </TextContent>
              )}
            </PageHeaderToolsItem>
          </FlexItem>
        </Flex>
      </MastheadMain>
    </Masthead>
  );
}
