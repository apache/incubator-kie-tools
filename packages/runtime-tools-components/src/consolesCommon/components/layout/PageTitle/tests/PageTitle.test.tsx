/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { PageTitle } from "../PageTitle";
import { Label } from "@patternfly/react-core/dist/js/components/Label";
import { render, screen } from "@testing-library/react";

describe("PageTitle test", () => {
  it("default snapshot testing", () => {
    const { container } = render(<PageTitle title="Title" />);

    expect(container).toMatchSnapshot();
  });

  it("snapshot testing with extra", () => {
    const { container } = render(<PageTitle title="Title" extra={<Label>Label</Label>} />);

    expect(container).toMatchSnapshot();

    const extra = screen.getAllByText("Label");

    expect(extra).toBeTruthy();
  });
});
