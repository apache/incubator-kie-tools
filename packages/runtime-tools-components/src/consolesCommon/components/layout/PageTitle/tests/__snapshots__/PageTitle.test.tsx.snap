// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle test default snapshot testing 1`] = `
<div>
  <div
    class="pf-l-flex"
    data-ouia-component-type="page-title"
    data-ouia-safe="true"
  >
    <div
      class="pf-m-spacer-sm"
    >
      <h1
        class="pf-c-title pf-m-4xl"
        data-ouia-component-id="OUIA-Generated-Title-1"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Title
      </h1>
    </div>
  </div>
</div>
`;

exports[`PageTitle test snapshot testing with extra 1`] = `
<div>
  <div
    class="pf-l-flex"
    data-ouia-component-type="page-title"
    data-ouia-safe="true"
  >
    <div
      class="pf-m-spacer-sm"
    >
      <h1
        class="pf-c-title pf-m-4xl"
        data-ouia-component-id="OUIA-Generated-Title-2"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Title
      </h1>
    </div>
    <div
      class="pf-m-spacer-sm"
    >
      <span
        class="pf-c-label"
      >
        <span
          class="pf-c-label__content"
        >
          Label
        </span>
      </span>
    </div>
  </div>
</div>
`;
